### PR TITLE
build: update to node 20

### PR DESCRIPTION
**Pull request by crisbeto**
Updates the repo to Node 20 and resolves some build issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,8 +24,8 @@
     "/ ": "",
     "postinstall": "node --preserve-symlinks --preserve-symlinks-main ./tools/postinstall-patches.js && patch-package --patch-dir tools/esm-interop/patches/npm && patch-package --patch-dir tools/npm-patches",
     "prepare": "husky",
-    "ng-dev": "ts-node --esm --project .ng-dev/tsconfig.json --transpile-only node_modules/@angular/ng-dev/bundles/cli.mjs",
-    "build": "ts-node --esm --project scripts/tsconfig.json scripts/build/build-packages-dist.mts",
+    "ng-dev": "node --no-warnings=ExperimentalWarning --loader ts-node/esm/transpile-only node_modules/@angular/ng-dev/bundles/cli.mjs",
+    "build": "node --no-warnings=ExperimentalWarning --loader ts-node/esm/transpile-only scripts/build/build-packages-dist.mts",
```
Comment (crisbeto) [inline package.json:28]: There's a comment in the tsconfig around why we need to it this way.

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-18.18.2
+20.9.0
```
Comment (jessicajaniuk) [inline .nvmrc:1]: Just curious why 20.9.0 and not 20.11.0 since the current [LTS](https://nodejs.org/en) is 20.11.0.

Comment (crisbeto) [reply .nvmrc]: I've changed it to 20.11.0.

**Comment by crisbeto** [comment]
Parking this PR until v18 as discussed.

File: WORKSPACE [hunk 3]
```diff
@@ -64,17 +64,17 @@ load("@rules_nodejs//nodejs:repositories.bzl", "nodejs_register_toolchains")
...
+        "20.11.0-darwin_amd64": ("node-v20.11.0-darwin-x64.tar.gz", "node-v20.11.0-darwin-x64", "c0ba02c905814258bd99a362027f8d4d2cc738218a9cf1dce2620e8735e3a80e"),
+        "20.11.0-linux_arm64": ("node-v20.11.0-linux-arm64.tar.xz", "node-v20.11.0-linux-arm64", "f6df68c6793244071f69023a9b43a0cf0b13d65cbe86d55925c28e4134d9aafb"),
+        "20.11.0-linux_ppc64le": ("node-v20.11.0-linux-ppc64le.tar.xz", "node-v20.11.0-linux-ppc64le", "6a0e1fa23d7bc707711bbc36159b4220eca123e13435d266d690c6b6c443dc67"),
+        "20.11.0-linux_s390x": ("node-v20.11.0-linux-s390x.tar.xz", "node-v20.11.0-linux-s390x", "cc92efa3fa101d613539451b1cf323ea9ac6198b4a68a7d3bf3b1090c6a7b5da"),
+        "20.11.0-linux_amd64": ("node-v20.11.0-linux-x64.tar.xz", "node-v20.11.0-linux-x64", "822780369d0ea309e7d218e41debbd1a03f8cdf354ebf8a4420e89f39cc2e612"),
+        "20.11.0-windows_amd64": ("node-v20.11.0-win-x64.zip", "node-v20.11.0-win-x64", "893115cd92ad27bf178802f15247115e93c0ef0c753b93dca96439240d64feb5"),
     },
-    # We need at least Node 18.17 due to some transitive dependencies.
-    node_version = "18.18.2",
+    node_version = "20.11.0",
```
Comment (JeanMeche) [inline WORKSPACE:77]: FYI, the CLI requires `>= 20.11.1` right now. [See](https://github.com/angular/angular-cli/blob/main/package.json)

**Pull request closed without merging**